### ISSUE TITLE
Interference of gestures (GestureRecognizer and several Touches)

### ISSUE THREAD

**Issue by mooseDev** [gloubibou/HHPanningTableViewCell#4]
Hi,

I really like your approach but I got some problems:
1. If you use a 2-finger (or more) swipe gesture on tableview  it is possible to reveal several drawer views at the time ... 
2.  I have a 2 finger swipe gesture on self.view but I can't manage to make HHPanningTableViewCell (tableView) to react only on one touch - currently my gesture on self.view and on (several) HHPanningTableViewCell will fire at the same time

Basically both problems are connected due the interference of swipe gestures...

**Comment by mooseDev** [comment]
I tried the following in HHPanningTableViewCell.m:
- (BOOL)gestureRecognizer:(UIGestureRecognizer_)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer_)otherGestureRecognizer
  {
  if ([gestureRecognizer isKindOfClass:[HHDirectionPanGestureRecognizer class]]) {
  
  ```
  if ([otherGestureRecognizer isKindOfClass:[UIPanGestureRecognizer class]]) {
      if(((UIPanGestureRecognizer*)otherGestureRecognizer).numberOfTouches > 1){
          [gestureRecognizer requireGestureRecognizerToFail:otherGestureRecognizer];
          return YES;
      }
  }
  
  HHDirectionPanGestureRecognizer *panGestureRecognizer = (HHDirectionPanGestureRecognizer*)gestureRecognizer;
  return (!panGestureRecognizer.panRecognized);
  ```
  
  }
  
  return YES;
  }

No more simultaneous reveals of drawer views and my 2 finger swipe on self.view works perfect...
But after first 2finger swipe HHPanningTableViewCells won't work like before...

Any suggestions?

**Comment by gloubibou** [comment]
I don't think the delegate is the right place to link both gesture recognizers.

Please try creating the dependency t the moment you add your own gesture recognizer.

**Comment by mooseDev** [comment]
I 'll give it a try... But what about general restriction of 2 or more finger swipe like mentioned in:
1. If you use a 2-finger (or more) swipe gesture on tableview it is possible to reveal several drawer views at the time ...

**Comment by gloubibou** [comment]
Not sure.

Try updating [HHPanningTableViewCell gestureRecognizer:shouldReceiveTouch:]  to return NO if

tableView.isDragging || tableView.isDecelerating || tabListController.isTabListRevealed

**Comment by mooseDev** [comment]
purging current cell when recognizing other gestures and redrawing was the solution
